tests/ttnn/unit_tests/operations/ccl/perf/perf_csv.py

```diff
@@ -52,70 +57,175 @@ def clean_attributes(attributes):
-        "DEVICE FW DURATION [ns]",
-        "DEVICE KERNEL DURATION [ns]",
-    ]
-    filtered_df = filtered_df[new_order]
 
     numeric_columns = [
         "HOST DURATION [ns]",
         "Cycles Count",
         "OP TO OP LATENCY [ns]",
         "DEVICE FW DURATION [ns]",
         "DEVICE KERNEL DURATION [ns]",
+        "DEVICE ERISC KERNEL DURATION [ns]",
     ]
     filtered_df[numeric_columns] = filtered_df[numeric_columns].apply(pd.to_numeric, errors="coerce")
 
     def calculate_min_avg_max_by_common_runs(df):
-        group_columns = ["ATTRIBUTES_BACKUP", "Input Shape", "dim", "num_links", "topology", "Layout", "Data Type"]
+        group_columns = [
+            "ATTRIBUTES_BACKUP",
+            "Input Shape",
+            "Output Shape",
+            "dim",
+            "num_links",
+            "topology",
+            "Layout",
+            "Data Type",
+            "OP CODE",
+        ]
         results = []
 
         for name, group in df.groupby(group_columns):
-            group_excluded = group.iloc[4:]
+            n_chips = group["n_chips"].iloc[0]
+            group_excluded = group.iloc[2 * n_chips :]
             if not group_excluded.empty:
                 min_values = group_excluded[numeric_columns].min().round(2)
                 avg_values = group_excluded[numeric_columns].mean().round(2)
                 max_values = group_excluded[numeric_columns].max().round(2)
 
+                op_bw_values, link_bw_values = [], []
+
+                for _, row in group_excluded.iterrows():
+                    op_bw, link_bw = calculate_bandwidth(row)
+                    op_bw_values.append(op_bw)
+                    link_bw_values.append(link_bw)
+
+                op_bw_min, op_bw_avg, op_bw_max = (
+                    round(min(op_bw_values), 2),
+                    round(sum(op_bw_values) / len(op_bw_values), 2),
+                    round(max(op_bw_values), 2),
+                )
+                link_bw_min, link_bw_avg, link_bw_max = (
+                    round(min(link_bw_values), 2),
+                    round(sum(link_bw_values) / len(link_bw_values), 2),
+                    round(max(link_bw_values), 2),
+                )
+
                 result_row = {
                     **{col: f"{min_values[col]} - {avg_values[col]} - {max_values[col]}" for col in numeric_columns},
+                    "Op BW [GB/s]": f"{op_bw_min} - {op_bw_avg} - {op_bw_max}",
+                    "Link BW [GB/s]": f"{link_bw_min} - {link_bw_avg} - {link_bw_max}",
                     **{key: value for key, value in zip(group_columns, name)},
+                    "n_chips": n_chips,
                 }
                 results.append(result_row)
 
         return pd.DataFrame(results)
 
-    average_values_by_common_run = calculate_min_avg_max_by_common_runs(filtered_df)
+    def calculate_bandwidth(row):
+        dtype = row["Data Type"]
+        dtype_sizes = {
+            "BFLOAT16": 2,
+            "BFLOAT8_B": 1,
```

Comment:
since we are adding the element size, then block floating point formats will be incorrectly reported since their size isn't a clean size per element of one (for example, bfp8 is 1088 B per tile of 1024 elements).

For those formats you'd need to calculate the tensor volume differently.